src/differentials/abstract_differential.jl

```diff
@@ -31,10 +36,24 @@ Base.:+(x::AbstractDifferential) = x
 """
     extern(x)
 
-Return `x` converted to an appropriate non-`AbstractDifferential` type, for use
-with external packages that might not handle `AbstractDifferential` types.
+Makes a best effort attempt to convert a differential into a primal value.
+This is not always a well-defined operation.
+For two reasons:
+ - It may not be possible to determine the primal type for a given differential.
+ For example `Zero` is a valid differential for any primal.
```

Comment:
Keeps it consistent with the other reasons below.

```suggestion
 For example, `Zero` is a valid differential for any primal.
```